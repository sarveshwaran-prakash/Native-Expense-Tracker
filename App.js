import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { createMaterialBottomTabNavigator } from "@react-navigation/material-bottom-tabs";
import AddTaskScreen from "./screens/AddTaskScreen";
import ViewTasksScreen from "./screens/ViewTasksScreen";
import { TaskProvider } from "./store/TaskContext";
import { Text } from "react-native";
import Icon from "react-native-vector-icons/FontAwesome"; // for FontAwesome icons

const Tab = createMaterialBottomTabNavigator();

export default function App() {
  return (
    <TaskProvider>
      <NavigationContainer>
        <Tab.Navigator>
          <Tab.Screen
            name="Add Task"
            component={AddTaskScreen}
            options={{
              tabBarLabel: <Text>ADD TASKS</Text>,
              tabBarIcon: ({ color }) => (
                <Icon name="plus" color={color} size={26} />
              ),
            }}
          />
          <Tab.Screen
            name="View Tasks"
            component={ViewTasksScreen}
            options={{
              tabBarLabel: <Text>VIEW TASKS</Text>,
              tabBarIcon: ({ color }) => (
                <Icon name="list" color={color} size={26} />
              ),
            }}
          />
        </Tab.Navigator>
      </NavigationContainer>
    </TaskProvider>
  );
}
