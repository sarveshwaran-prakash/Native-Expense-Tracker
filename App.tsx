import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
import { ExpenseProvider } from "./store/ExpenseContext";
import AddExpenseScreen from "./screens/AddExpenseScreen";
import ViewExpensesScreen from "./screens/ViewExpensesScreen";
import Icon from "react-native-vector-icons/FontAwesome";

const Tab = createBottomTabNavigator();

export default function App() {
  return (
    <ExpenseProvider>
      <NavigationContainer>
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === "Add Transactions") {
                iconName = "plus";
              } else if (route.name === "View Transactions") {
                iconName = "list";
              }

              if (iconName) {
                return <Icon name={iconName} color={color} size={26} />;
              }

              return null;
            },
            tabBarActiveTintColor: "#3F72AF",
            tabBarInactiveTintColor: "gray",
            tabBarStyle: {
              backgroundColor: "#DBE2EF",
            },
            tabBarLabelStyle: {
              color: "black",
              fontSize: 15,
              fontWeight: "600",
            },
          })}
        >
          <Tab.Screen
            name="Add Transactions"
            component={AddExpenseScreen}
            options={() => ({
              headerStyle: { backgroundColor: "#DBE2EF" },
            })}
          />
          <Tab.Screen
            name="View Transactions"
            component={ViewExpensesScreen}
            options={() => ({
              headerStyle: { backgroundColor: "#DBE2EF" },
            })}
          />
        </Tab.Navigator>
      </NavigationContainer>
    </ExpenseProvider>
  );
}
