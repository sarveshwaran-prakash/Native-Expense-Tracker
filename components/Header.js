import React from "react";
import { View, Text, StyleSheet } from "react-native";

export default function Header({ title }) {
  return (
    <View style={styles.header}>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    marginTop: 40,
    width: "100%",
    height: 60,
    backgroundColor: "grey",
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    color: "white",
    fontSize: 20,
    fontWeight: "bold",
  },
});
