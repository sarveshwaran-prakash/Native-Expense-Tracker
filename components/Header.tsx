import React from "react";
import { View, Text, StyleSheet } from "react-native";

interface HeaderProps {
  title: string;
}

const Header: React.FC<HeaderProps> = ({ title }: HeaderProps) => {
  return (
    <View style={styles.header} testID="header-container">
      <Text style={styles.title}>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    marginTop: 40,
    width: "100%",
    height: 60,
    backgroundColor: "grey",
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    color: "white",
    fontSize: 20,
    fontWeight: "bold",
  },
});

export default Header;
