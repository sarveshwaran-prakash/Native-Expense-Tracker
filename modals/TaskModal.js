// TaskModal.js
import React, { useState } from "react";
import { Modal, View, TouchableOpacity, Text, StyleSheet } from "react-native";
import { FontAwesome } from "@expo/vector-icons";
import EditTaskModal from "./EditTaskModal";

export default function TaskModal({
  visible,
  onClose,
  onEdit,
  onDelete,
  task,
  description,
}) {
  const [editModalVisible, setEditModalVisible] = useState(false);

  const handleEdit = () => {
    setEditModalVisible(true);
  };

  const handleSaveEdit = (editedTask, editedDescription) => {
    onEdit(editedTask, editedDescription);
  };

  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <TouchableOpacity
        style={styles.container}
        activeOpacity={1}
        onPress={onClose}
      >
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <TouchableOpacity onPress={handleEdit}>
              <Text style={styles.modalOption}>
                <FontAwesome name="edit" size={20} />
                Edit Task
              </Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onDelete}>
              <Text style={styles.modalOption}>
                <FontAwesome name="trash" size={20} /> Delete Task
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </TouchableOpacity>
      <EditTaskModal
        visible={editModalVisible}
        onClose={() => setEditModalVisible(false)}
        task={task}
        description={description}
        onSave={handleSaveEdit}
      />
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "rgba(0, 0, 0, 0.5)",
    justifyContent: "center",
    alignItems: "center",
  },
  centeredView: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalView: {
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  modalOption: {
    marginBottom: 20,
  },
});
