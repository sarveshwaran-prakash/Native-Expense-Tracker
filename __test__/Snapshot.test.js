import React from "react";
import App from "../App";
import renderer from "react-test-renderer";
import Header from "../components/Header";
import TaskModal from "../modals/TaskModal";

test("State Provider snapshot", () => {
  const snap = renderer.create(<TaskModal />).toJSON();
  expect(snap).toMatchSnapshot();
});
