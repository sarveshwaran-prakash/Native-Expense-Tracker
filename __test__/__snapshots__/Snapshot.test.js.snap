// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State Provider snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "grey",
      "height": 60,
      "justifyContent": "center",
      "marginTop": 40,
      "width": "100%",
    }
  }
  testID="header-container"
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  />
</View>
`;
